integration_files/SweRV_EH1/testbench/test_tb_top.cpp

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2019 Western Digital Corporation or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include <stdlib.h>
+#include <iostream>
+#include <utility>
+#include <string>
+#include <sstream>
+#include "Vtb_top.h"
+#include "verilated.h"
+#include "verilated_vcd_c.h"
+
+vluint64_t main_time = 0;
+
+double sc_time_stamp () {
+ return main_time;
+}
+
+
+int main(int argc, char** argv) {
+  std::cout << "\nVerilatorTB: Start of sim\n" << std::endl;
+
+  Verilated::commandArgs(argc, argv);
+
+  Vtb_top* tb = new Vtb_top;
+
+  // init trace dump
+  VerilatedVcdC* tfp = NULL;
+  
+  // initializing file path for sim.vcd
+  std::stringstream file_path;
+  std::string file;
+
+  // appending sim_dir path with sim.vcd to copy it in the right folder
+  file_path << argv[3] << "/sim.vcd";
```

Comment:
Looks good. However can you rename agrv[3] to path? so that anyone will be able to see what is happening.

Reply:
Will just printing the path or argv[3] work? As for in future to see if it changes. We can do this while making the generic testbench in cpp.

Reply:
Can you open a new issue for this ?

Reply:
Will open an issue for this.